data_pipeline_api/registry/access_upload.py

```diff
@@ -328,73 +391,79 @@ def upload_model_run(
         data_directory = config_filename.parent / data_directory
     run_id = config["run_id"]
     config_yaml = config["config"]
-    responsible_person = _create_target_data_dict(
-        DataRegistryTarget.users, {DataRegistryField.username: config["responsible_person"]}
-    )
+    accessibility = _get_accessibility(config)
+    namespace = config.get("namespace")
     model_version_str = config_yaml["model_version"]
     model_name = config_yaml["model_name"]
-    accessibility = _create_target_data_dict(
-        DataRegistryTarget.accessibility, {DataRegistryField.name: accessibility_name}
-    )
     open_timestamp = config_yaml["open_timestamp"]
+    model_git_sha = config["metadata"]["git_sha"]
+    model_uri = config["metadata"]["uri"]
+
     inputs = []
     outputs = []
     posts = []
 
-    _, storage_root = _add_storage_type_and_root(posts, remote_uri_override, data_registry_url, token)
+    storage_root = _add_storage_root(posts, remote_uri_override, accessibility, data_registry_url, token)
+    code_repo = _add_code_repo(posts, model_name, model_version_str, model_git_sha, model_uri)
```

Comment:
It's not clear to me whether we expect to be doing this. From some conversation on zulip I got the impression that this would be done after the fact.

Reply:
my impression from zulip was that we should be doing it! It's not a huge deal either way

Reply:
Yeah, I might have just been confused. My understanding was that the coderepo/version stuff would happen independently from the upload, but I'm really not sure. Anyway, as you suggest, it is easy to change.